New Topic: C# with a few initial concepts and terms (#1017)
Codecademy/docs: content/c-sharp/concepts/arrays/arrays.md

---
Title: 'Arrays'
Description: 'An array is a data structure used in C# to store a sequential collection of elements.'
Subjects:
  - 'Computer Science'
  - 'Code Foundations'
Tags:
  - 'Arrays'
CatalogContent:
  - 'learn-c-sharp'
  - 'paths/computer-science'
---

> An **array** is a data structure used in C# to store a sequential collection of elements. Its size cannot be changed after creation. The elements of an array are all of the same type, but it is possible to define a C# array that can hold elements of any type by specifying the type of the array as `object`. In C#, all types directly or indirectly inherit from the Object type.

Reviewer: ```suggestion
An **array** is a data structure used in [C#](https://www.codecademy.com/resources/docs/c-sharp) to store a sequential collection of elements. Its size is immutable (cannot be changed after creation). The elements of an array are all of the same type, but it is possible to define a C# array that can hold elements of any type by specifying the type of the array as `object`. In C#, all types directly or indirectly inherit from the `Object` type.
```

> An **array** is a data structure used in [C#](https://www.codecademy.com/resources/docs/c-sharp) to store a sequential collection of elements. Its size is immutable (cannot be changed after creation). The elements of an array are all of the same type, but it is possible to define a C# array that can hold elements of any type by specifying the type of the array as an object. In C#, all types directly or indirectly inherit from the `object` type.

Reviewer 2: ```suggestion
An **array** is a data structure used in [C#](https://www.codecademy.com/resources/docs/c-sharp) to store a sequential collection of elements. Its size is immutable (cannot be changed after creation). The elements of an array are all of the same type, but it is possible to define a C# array that can hold elements of any type by specifying the type of the array as an `object`. In C#, all types directly or indirectly inherit from the `object` type.
```

> An **array** is a data structure used in [C#](https://www.codecademy.com/resources/docs/c-sharp) to store a sequential collection of elements. Its size is immutable (cannot be changed after creation). The elements of an array are all of the same type, but it is possible to define a C# array that can hold elements of any type by specifying the type of the array as an `object`. In C#, all types directly or indirectly inherit from the `object` type.

Reviewer: ```suggestion
An **array** is a data structure used in [C#](https://www.codecademy.com/resources/docs/c-sharp) to store a sequential collection of elements. Its size is immutable (cannot be changed after creation). The elements of an array are all of the same type, but it is possible to define a C# array that can hold elements of any type by specifying the type of the array as an object. In C#, all types directly or indirectly inherit from the `Object` type.
```

## Syntax

> There are several different ways to create an array in C#:

Reviewer 2: ```suggestion
There are several ways to create an array in C#:
```

```pseudo
// Create a variable of type "type[]" without initalizing it:
type[] arrayName;

> // Create the array varaible and initalize it with an array of N items:

Reviewer: ```suggestion
// Create the array variable and initalize it with an array of N items:
```

type[] arrayName = new type[N];

// Create the array variable and intalize it by specifying the contents:
type[] arrayName = new type[] { value1, value2, value3, ... valueN };

// Alternative way of creating the array and specifying the contents:
type[] arrayName = { value1, value2, value3, ... valueN };
```

> Arrays in C++ have a set size, meaning the number of elements it holds cannot be changed once the array has been created.

Reviewer: Maybe this can be made into a note as to not confuse readers.

## Example

> Each element in an array is assigned a specific index starting at zero. To access or modify an element in the array you may simply refer to it by its index and operate on it accordingly.

Reviewer 2: ```suggestion
Each element in an array is assigned a specific index starting at zero. To access or modify an element in the array, you refer to it by its index and operate on it accordingly.
```

```cs
using System;

public class Example
{
  public static void Main(string[] args)
  {
    char[] vowels = {'a', 'e', 'i', 'o', 'u'};
    //      indexes:  0    1    2    3    4

    Console.WriteLine(vowels[0]); // Output: a

    vowels[0] = 'r';

    Console.WriteLine(vowels[0]); // Output: r
  }
}
```

> In the case above, an array of chars was initialized with all the vowels, and then the first element in the array at index 0 was printed. Then the element at index 0 was modified by assigning it a new value of `'r'`, which was printed out below.

Reviewer: ```suggestion
In the case above, an array of `char`s was initialized with all the vowels, and the first element in the array at index 0 was printed. Then, the element at index 0 was modified by assigning it a new value of `'r'`, which was printed out below.
```

> In the case above, an array of `char`s was initialized with all the vowels, and the first element in the array at index 0 was printed. Then, the element at index 0 was modified by assigning it a new value of `'r'`, which was printed out below.

Reviewer 2: ```suggestion
In the example above, an array of `char`s was initialized with all the vowels. The first element in the array at index 0 was printed. Then, the element at index 0 was modified by assigning it a new value of `'r'`. The value at index 0 was printed again and the results are shown below.
```

> In the example above, an array of `char`s was initialized with all the vowels. The first element in the array at index 0 was printed. Then, the element at index 0 was modified by assigning it a new value of `'r'`. The value at index 0 was printed again and the results are shown below.

Reviewer: ```suggestion
In the example above, an array of `char`s was initialized with all the vowels. The first element in the array at index 0 was printed. Then, the element at index 0 was modified by assigning it a new value of `'r'`. Then, the value at index 0 was printed again.
```

## Array Methods

> Arrays in C# are objects, not just contiguous blocks of memory as in C and C++. `Array` is the base type of all array types, and any array can use the properties and methods of the `Array` object. A few of which are listed below:

Reviewer 2: ```suggestion
Arrays in C# are objects, not just contiguous blocks of memory as in C and C++. `Array` is the base type of all arrays, and any array can use the properties and methods of the `Array` object,  a few of which are listed below:
```
